### PR TITLE
[17.0][MIG] stock_request: Migration to 17.0

### PR DESCRIPTION
**Pull request by celm1990**

### CODE REVIEW

**Comment by celm1990** [comment]
@JordiBForgeFlow @pedrobaeza could you please take a look at this PR?

**Review by smcqueen81 — COMMENTED** [review]
Functional ok

**Comment by celm1990** [comment]
squash done

**Comment by ygol** [comment]
I am unable to create a request (following the usage instructions) on runboat. 
Is there some settings I am missing?

image

Thank you

**Comment by celm1990** [comment]
> I am unable to create a request (following the usage instructions) on runboat. Is there some settings I am missing?
> 



@ygol  I updated the code, please try again

**Comment by celm1990** [comment]
@OCA/logistics-maintainers , could you please help me review this PR and merge it if it is OK?

**Comment by ygol** [comment]
LGTM functionally

**Comment by celm1990** [comment]
@pedrobaeza could you help me with merge if it's OK

**Comment by pedrobaeza** [comment]
It requires reviews.

**Comment by celm1990** [comment]
ping @OCA/logistics-maintainers

**Review by jbaudoux — APPROVED** [review]
LG

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by celm1990** [comment]
@OCA/logistics-maintainers could somebody help me with review/merge, please

**Comment by celm1990** [comment]
@OCA/logistics-maintainers could somebody help me with review/merge, please

File: stock_request/models/stock_request.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def _get_default_requested_by(self):
     def _get_expected_date():
         return fields.Datetime.now()
 
-    name = fields.Char(states={"draft": [("readonly", False)]})
+    name = fields.Char(readonly=True)
```
Comment (pedrobaeza) [inline stock_request/models/stock_request.py]: Please remove `readonly=True` here, as it won't be importable if not, and put it in the view (with the `state` condition).

**Review by waoconnect-ravi — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
See my previous comment

File: stock_request/models/stock_request.py (hunk 1 above)
Comment (celm1990) [reply stock_request/models/stock_request.py]: @pedrobaeza , I removed the `readonly` attribute from the Python code, as it already exists in the view as `readonly="1"`. I believe this was causing confusion and wasn't working in the previous version, as this field should always be readonly.
Please let me know if the change is OK.

img from V16
image

**Comment by pedrobaeza** [comment]
@celm1990 it seems you haven't put the readonly condition in the view.

**Review by pedrobaeza — APPROVED** [review]
Forget my last comment that have arrived at the same time:

/ocabot merge nobump

**Pull request merged**